$NetBSD: patch-src_CMakeLists.txt,v 1.7 2024/10/13 14:13:23 gdt Exp $

The first hunk avoids some cmake magic, which buggily adds -ldl on NetBSD.
\todo File a bug against pkgsrc websockets, perhaps cascading to upstream.

The second hunk removes linker args on SunOS, for no apparent reason.
\todo Explain and file a bug upstream; this should not be a pkgsrc change.

--- src/CMakeLists.txt.orig	2020-08-19 13:55:03.000000000 +0000
+++ src/CMakeLists.txt
@@ -200,7 +200,7 @@
 			link_directories(${mosquitto_SOURCE_DIR})
 		endif (WIN32)
 	else (STATIC_WEBSOCKETS)
-		set (MOSQ_LIBS ${MOSQ_LIBS} websockets_shared)
+		set (MOSQ_LIBS ${MOSQ_LIBS} websockets)
 	endif (STATIC_WEBSOCKETS)
 endif (WITH_WEBSOCKETS)
 
@@ -213,9 +213,9 @@ target_link_libraries(mosquitto ${MOSQ_L
 if (UNIX)
 	if (APPLE)
 		set_target_properties(mosquitto PROPERTIES LINK_FLAGS "-Wl,-exported_symbols_list -Wl,${mosquitto_SOURCE_DIR}/src/linker-macosx.syms")
-	else (APPLE)
+	elseif (NOT CMAKE_SYSTEM_NAME STREQUAL SunOS)
 		set_target_properties(mosquitto PROPERTIES LINK_FLAGS "-Wl,-dynamic-list=${mosquitto_SOURCE_DIR}/src/linker.syms")
-	endif (APPLE)
+	endif ()
 endif (UNIX)
 
 install(TARGETS mosquitto RUNTIME DESTINATION "${CMAKE_INSTALL_SBINDIR}")
